src/sentry/snuba/outcomes.py

```diff
@@ -263,40 +271,58 @@ def __init__(
             query_groupby.update(groupby.get_snuba_groupby())
         self.query_groupby = list(query_groupby)
 
-        self.conditions, self.filter_keys = get_filter(query, params)
+        self.group_by = []
+        for key in self.query_groupby:
+            self.group_by.append(Column(key))
+
+        conditions, filter_keys = get_filter(query, params)
+
+        for org in filter_keys.get("org_id"):
+            self.conditions.append(
+                Condition(Column("org_id"), Op.EQ, org),
+            )
+        if filter_keys.get("project_id") is not None:
+            self.conditions.append(
+                Condition(Column("project_id"), Op.IN, filter_keys.get("project_id")),
+            )
+        for condition in conditions:
+            self.conditions.append(Condition(Column(condition[0]), Op.IN, condition[2]))
+
+        self.select_params = []
+        for aggregation in aggregations:
+            aggregation_field = aggregation[2] if aggregation[1] == "" else aggregation[1]
```

Comment:
mind turning this into a private function and adding a comment? not sure what this is doing